pkg/cloud/aws/actuators/machine/instances.go

```diff
@@ -250,6 +292,9 @@ func launchInstance(machine *clusterv1.Machine, machineProviderConfig *providerc
 		Placement:          placement,
 	}
 
+	if len(blockDeviceMappings) > 0 {
```

Comment:
You can skip the check given `getBlockDeviceMappings` always returns valid list of block device mappings (including nil = empty array).